jsk_2016_01_baxter_apc/node_scripts/work_order.py

```diff
@@ -5,24 +5,29 @@
 import json
 import rospy
 from jsk_2015_05_baxter_apc.msg import WorkOrder, WorkOrderArray
-from jsk_apc2016_common import get_bin_contents, get_work_order
+from jsk_apc2016_common import get_bin_contents, get_work_order, get_object_data
 
 
-def get_sorted_work_order(json_file):
+def get_sorted_work_order(json_file, apc2016):
     """Sort work order to maximize the score"""
+    gripper = rospy.get_param('~gripper', 'gripper2015')
     bin_contents = get_bin_contents(json_file=json_file)
-    bin_n_contents = dict(map(lambda (bin_, objects): (bin_, len(objects)), bin_contents.iteritems()))
-    sorted_work_order = []
     work_order = get_work_order(json_file=json_file)
-    for bin_, n_contents in sorted(bin_n_contents.items(), key=lambda x: x[1]):
-        if n_contents > 5:  # Level3
-            continue
-        sorted_work_order.append((bin_, work_order[bin_]))
+    object_data = get_object_data()
+    sorted_bin_list = bin_contents.keys()
+    if apc2016:
+        sorted_bin_list = sorted(sorted_bin_list,
+                                 key=lambda bin_: filter(lambda x: x['name'] == work_order[bin_],
+                                                         object_data)[0]['graspability'][0][gripper])
```

Comment:
Unreadable.

`filter(...)` can be rewritten using Python's list comprehension.

```
filter([x['name'] == work_order[bin_] for x in object_data])
```